cmd/operator-sdk/scorecard/cmd.go

```diff
@@ -15,55 +15,173 @@
 		log.Fatalf("Unable to add kubeconfig :%v", err)
 	}
-	if err := viper.BindPFlag("scorecard."+scorecard.OutputFormatOpt, scorecardCmd.Flags().Lookup(scorecard.OutputFormatOpt)); err != nil {
+	if err := viper.BindPFlag("scorecard."+outputFormatOpt, scorecardCmd.Flags().Lookup(outputFormatOpt)); err != nil {
 		log.Fatalf("Unable to add output format :%v", err)
 	}
 	if err := viper.BindPFlag("scorecard."+schelpers.VersionOpt, scorecardCmd.Flags().Lookup(schelpers.VersionOpt)); err != nil {
 		log.Fatalf("Unable to add version :%v", err)
 	}
-	if err := viper.BindPFlag("scorecard."+scorecard.SelectorOpt, scorecardCmd.Flags().Lookup(scorecard.SelectorOpt)); err != nil {
+	if err := viper.BindPFlag("scorecard."+selectorOpt, scorecardCmd.Flags().Lookup(selectorOpt)); err != nil {
 		log.Fatalf("Unable to add selector :%v", err)
 	}
-	if err := viper.BindPFlag("scorecard."+scorecard.ListOpt, scorecardCmd.Flags().Lookup(scorecard.ListOpt)); err != nil {
+	if err := viper.BindPFlag("scorecard."+listOpt, scorecardCmd.Flags().Lookup(listOpt)); err != nil {
 		log.Fatalf("Unable to add list :%v", err)
 	}
-	if err := viper.BindPFlag("scorecard."+scorecard.BundleOpt, scorecardCmd.Flags().Lookup(scorecard.BundleOpt)); err != nil {
+	if err := viper.BindPFlag("scorecard."+bundleOpt, scorecardCmd.Flags().Lookup(bundleOpt)); err != nil {
 		log.Fatalf("Unable to add bundle :%v", err)
 	}
+
 	return scorecardCmd
 }
+
+func initConfig() (*viper.Viper, error) {
+	// viper/cobra already has flags parsed at this point; we can check if a config file flag is set
+	if viper.GetString(configOpt) != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(viper.GetString(configOpt))
+	} else {
+		viper.AddConfigPath(projutil.MustGetwd())
+		// using SetConfigName allows users to use a .yaml, .json, or .toml file
+		viper.SetConfigName(scorecard.DefaultConfigFile)
+	}
+
+	var scViper *viper.Viper
+	if err := viper.ReadInConfig(); err == nil {
+		scViper = viper.Sub("scorecard")
+		// this is a workaround for the fact that nested flags don't persist on viper.Sub
+		scViper.Set(outputFormatOpt, viper.GetString("scorecard."+outputFormatOpt))
+		scViper.Set(scplugins.KubeconfigOpt, viper.GetString("scorecard."+scplugins.KubeconfigOpt))
+		scViper.Set(schelpers.VersionOpt, viper.GetString("scorecard."+schelpers.VersionOpt))
+		scViper.Set(selectorOpt, viper.GetString("scorecard."+selectorOpt))
+		scViper.Set(bundleOpt, viper.GetString("scorecard."+bundleOpt))
+		scViper.Set(listOpt, viper.GetString("scorecard."+listOpt))
+		// configure logger output before logging anything
+		if !scViper.IsSet(outputFormatOpt) {
+			scViper.Set(outputFormatOpt, scorecard.TextOutputFormat)
+		}
+		format := scViper.GetString(outputFormatOpt)
+		if format == scorecard.TextOutputFormat {
+			logReadWriter = os.Stdout
+		} else if format == scorecard.JSONOutputFormat {
+			logReadWriter = &bytes.Buffer{}
+		} else {
+			return nil, fmt.Errorf("invalid output format: %s", format)
+		}
```

Comment:
Nit: switch/case

Reply:
fixed in PR https://github.com/operator-framework/operator-sdk/pull/2428